src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.Windows.cs

```diff
@@ -0,0 +1,148 @@
+
+			platformView.CoreWebView2.Settings.AreDevToolsEnabled = true;//EnableWebDevTools;
+			platformView.CoreWebView2.Settings.IsWebMessageEnabled = true;
+			platformView.CoreWebView2.AddWebResourceRequestedFilter($"{AppOrigin}*", CoreWebView2WebResourceContext.All);
+			platformView.CoreWebView2.WebResourceRequested += CoreWebView2_WebResourceRequested;
+
+
+			platformView.Source = new Uri(new Uri(AppOriginUri, "/").ToString());
+
+		}
+
+		protected override void DisconnectHandler(WebView2 platformView)
+		{
+#if DEBUG
+			var logger = MauiContext!.Services!.GetService<ILogger<HybridWebViewHandler>>() ?? NullLogger<HybridWebViewHandler>.Instance;
+			logger.LogInformation("HybridWebViewHandler: Disconnecting WebView2");
+#endif
+
+			platformView.WebMessageReceived -= OnWebMessageReceived;
+			platformView.Close();
+
+			base.DisconnectHandler(platformView);
+		}
+
+		public static void MapSendRawMessage(IHybridWebViewHandler handler, IHybridWebView hybridWebView, object? arg)
+		{
+			if (arg is not string rawMessage || handler.PlatformView is not IHybridPlatformWebView hybridPlatformWebView)
+			{
+				return;
+			}
+
+			hybridPlatformWebView.SendRawMessage(rawMessage);
+		}
+
+
+		private void OnWebMessageReceived(WebView2 sender, CoreWebView2WebMessageReceivedEventArgs args)
+		{
+			VirtualView?.RawMessageReceived(args.TryGetWebMessageAsString());
+		}
+
+		private async void CoreWebView2_WebResourceRequested(CoreWebView2 sender, CoreWebView2WebResourceRequestedEventArgs eventArgs)
+		{
+			// Get a deferral object so that WebView2 knows there's some async stuff going on. We call Complete() at the end of this method.
+			using var deferral = eventArgs.GetDeferral();
+
+			var requestUri = HybridWebViewQueryStringHelper.RemovePossibleQueryString(eventArgs.Request.Uri);
+
+			if (new Uri(requestUri) is Uri uri && AppOriginUri.IsBaseOf(uri))
+			{
+				var relativePath = AppOriginUri.MakeRelativeUri(uri).ToString().Replace('/', '\\');
+
+				string contentType;
+				if (string.IsNullOrEmpty(relativePath))
+				{
+					relativePath = VirtualView.DefaultFile;
+					contentType = "text/html";
+				}
+				else
+				{
+					if (!ContentTypeProvider.TryGetContentType(relativePath, out contentType!))
+					{
+						// TODO: Log this
+						contentType = "text/plain";
+					}
+				}
+
+				var assetPath = Path.Combine(VirtualView.HybridRoot!, relativePath!);
+				var contentStream = await GetAssetStreamAsync(assetPath);
+
+				if (contentStream is null)
+				{
+					var notFoundContent = "Resource not found (404)";
+					eventArgs.Response = sender.Environment!.CreateWebResourceResponse(
+						Content: null,
+						StatusCode: 404,
+						ReasonPhrase: "Not Found",
+						Headers: GetHeaderString("text/plain", notFoundContent.Length)
+					);
+				}
+				else
+				{
+					eventArgs.Response = sender.Environment!.CreateWebResourceResponse(
+						Content: await CopyContentToRandomAccessStreamAsync(contentStream),
```

Comment:
Should we try catch any of this? Or use our FireAndForget and move this to a separate method?

Reply:
This is essentially copied from BlazorWebView so it's fairly tried-and-true. There's a fair bit of 'error' checking at each level, and there have been no reports of problems with the code (except now that I've jinxed it!).